dev/tasks/install-tools

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env bash
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+REPO_ROOT="$(git rev-parse --show-toplevel)"
+cd "${REPO_ROOT}"
+
+if [[ ! -x go ]]; then
+  echo "Must install go; assuming we are running in a container"
+  curl -L -o /tmp/go.tar.gz https://go.dev/dl/go1.22.4.linux-amd64.tar.gz
```

Comment:
I used to hit a controller-gen problem in go1.22 https://github.com/kubernetes-sigs/controller-tools/issues/880
If installing golang, can we install go1.23 instead?

Reply:
I think 1.22.4 is the latest: https://go.dev/doc/install

I think we updated controller-gen so we should be good with 1.22?

Reply:
you are absolutely right!